### PR TITLE
Fix and cleanup outgoing interface code

### PR DESCRIPTION
**Pull request by cas--**
There was a misunderstand about outgoing interface setting in libtorrent
and instead of being able to take both IP and adapater names, it only
accepts adapter names and errors with an IP address, which was the
default of '0.0.0.0' in code.

This fixes the code to not accept IP address and use empty string if it
is given one.

Also includes a bit of code cleanup.

### CODE REVIEW

File: deluge/core/preferencesmanager.py [hunk 1]
```diff
@@ -191,7 +191,11 @@ def _on_set_listen_interface(self, key, value):
         self.__set_listen_on()
 
     def _on_set_outgoing_interface(self, key, value):
-        self.__set_outgoing_on()
+        """ Set the interface address for outgoing BitTorrent connections."""
```
Comment (cas--) [inline deluge/core/preferencesmanager.py]: rename address to adapter name

File: deluge/ui/console/modes/preferences/preference_panes.py [hunk 2]
```diff
@@ -300,7 +300,7 @@ def create_pane(self, core_conf, console_config):
         self.add_header(_('Outgoing Interface'), space_above=True)
         self.add_text_input(
             'outgoing_interface',
-            _('IP address of the interface to open outgoing connections on. (leave empty for default):'),
+            _('The interface adapter name to open outgoing connections on. (leave empty for default):'),
```
Comment (cas--) [inline deluge/ui/console/modes/preferences/preference_panes.py]: Remove `The`

Comment (cas--) [reply deluge/ui/console/modes/preferences/preference_panes.py]: Interface adapter name for outgoing BitTorrent connections. (Leave  empty for default).

File: deluge/ui/gtkui/glade/preferences_dialog.ui [hunk 3]
```diff
@@ -2924,7 +2924,7 @@ used sparingly.</property>
                                       <object class="GtkEntry" id="entry_outgoing_interface">
                                         <property name="visible">True</property>
                                         <property name="can_focus">True</property>
-                                        <property name="tooltip_text" translatable="yes">IP address for outgoing BitTorrent connections. Leave this empty if you want to use the default.</property>
+                                        <property name="tooltip_text" translatable="yes">Interface adapter name for outgoing BitTorrent connections. Leave this empty if you want to use the default.</property>
```
Comment (cas--) [inline deluge/ui/gtkui/glade/preferences_dialog.ui]: `Interface adapter name for outgoing BitTorrent connections. (Leave  empty for default).`

File: deluge/ui/gtkui/preferences.py [hunk 4]
```diff
@@ -532,11 +532,12 @@ def set_config(self, hide=False):
...
         ).get_active()
         incoming_address = self.builder.get_object('entry_interface').get_text().strip()
-        outgoing_address = self.builder.get_object('entry_outgoing_interface').get_text().strip()
         if deluge.common.is_ip(incoming_address) or not incoming_address:
             new_core_config['listen_interface'] = incoming_address
-        if deluge.common.is_ip(outgoing_address) or not outgoing_address:
-            new_core_config['outgoing_interface'] = outgoing_address
+        outgoing_interface = self.builder.get_object(
+            'entry_outgoing_interface').get_text().strip()
+        if not deluge.common.is_ip(outgoing_address) or not outgoing_interface:
```
Comment (cas--) [inline deluge/ui/gtkui/preferences.py]: `outgoing_interface`

**Comment by doadin** [comment]
https://github.com/arvidn/libtorrent/blob/f0227451a6a7a28b31523172e7eb23bb6762b20c/include/libtorrent/string_util.hpp#L101

**Comment by cas--** [comment]
I wanted to fix the code quickly to what is written in the libtorrent docs
as it was broken for Linux.

You can make another PR to add Windows support. Perhaps some tests to
ensure that it is correct for Windows and Linux now that we have Appveyor.

Also ask Arvid to update docs

**Pull request merged**